send_review_notification/github_services.py

```diff
@@ -0,0 +1,266 @@
+            headers=_get_request_headers()
+        )
+        response.raise_for_status()
+        pr_subset = response.json()
+
+        if len(pr_subset) == 0:
+            break
+        page_number += 1
+
+        pull_requests = [
+            github_domain.PullRequest.from_github_response(pull_request)
+            for pull_request in pr_subset]
+        update_assignee_timestamp(org_name, repository, pull_requests)
+        for pull_request in pull_requests:
+            if not pull_request.is_reviewer_assigned():
+                continue
+            for reviewer in pull_request.assignees:
+                pending_review_time = (
+                    datetime.datetime.now(datetime.timezone.utc) -
+                    reviewer.timestamp)
+                if (reviewer.name != pull_request.author) and (
+                        pending_review_time >=
+                        datetime.timedelta(hours=wait_hours)
+                    ):
+                    reviewer_to_assigned_prs[reviewer.name].append(pull_request)
+    return reviewer_to_assigned_prs
+
+
+# Here we use type Any because the response we get from the api call is hard
+# to annotate in a typedDict, Hence used type Any.
+def __process_activity(
+    pull_request: github_domain.PullRequest,
+    event: Dict[str, Any]
+) -> None:
+    """Process activity and updates assignee timestamps."""
+    if event['event'] != 'assigned':
+        return
+
+    assignee = pull_request.get_assignee(event['assignee']['login'])
+    event_timestamp = parser.parse(event['created_at'])
+    if assignee:
+        assignee.set_timestamp(max([assignee.timestamp, event_timestamp]))
+
+
+def update_assignee_timestamp(
+    org_name: str,
+    repository: str,
+    pr_list: List[github_domain.PullRequest]
+) -> None:
+    """Fetches PR timeline and updates assignment timestamp."""
+    for pull_request in pr_list:
+        pr_number = pull_request.number
+        activity_url = ISSUE_TIMELINE_URL_TEMPLATE.format(
+            org_name, repository, pr_number)
+
+        page_number = 1
+        while True:
+            logging.info('Fetching PR #%s timeline', pr_number)
+            response = requests.get(
+                activity_url,
+                params={'page': page_number, 'per_page': 100},
+                headers={
+                    'Accept': 'application/vnd.github.mockingbird-preview+json',
+                    'Authorization': f'token {_TOKEN}'}
+            )
+            response.raise_for_status()
+            timeline_subset = response.json()
+
+            if len(timeline_subset) == 0:
+                break
+
+            for event in timeline_subset:
+                __process_activity(pull_request, event)
+
+            page_number += 1
+
+
+@check_token
+def create_discussion_comment(org_name: str, repo: str, body: str) -> None:
+    """Creates github discussion on the team_slug with the given title and
+    body.
+    """
+
+    query_category_id = """
```

Comment:
The `_id` suffix doesn't seem right here since this string isn't really an ID. Based on the [GitHub GraphQL API docs](https://docs.github.com/en/graphql/guides/forming-calls-with-graphql#working-with-variables), I think it's actually just called the query

ditto below

Reply:
changed it to  `query`